### PR TITLE
react1 week2 HW first commit

### PR DESCRIPTION
**Pull request by agkabir**

### CODE REVIEW

File: react/react1/week2/todo-list/src/Counter.jsx [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+function Counter() {
+    const [count, setCount] = useState(0);
+    useEffect(() => {
+      const timer =  setTimeout(() => {
+          setCount((prev) => prev + 1);
+      }, 1000);
+        return () => {
+          clearInterval(timer);
+        };
```
Comment (Divyajg) [inline react/react1/week2/todo-list/src/Counter.jsx:11]: Good job with timer.

File: react/react1/week2/todo-list/src/TodoItem.jsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+      spanElement.style.textDecoration = "line-through";
+    } else {
+      spanElement.style.textDecoration = "none";
+    }
+  }, [isChecked, todo.id]);
+
+  return (
+    <li>
+      <span id={`span${todo.id}`}>{todo.description}</span>
+      <input type="checkbox" checked={isChecked} onChange={changeOnChecked} />
```
Comment (Divyajg) [inline react/react1/week2/todo-list/src/TodoItem.jsx:21]: It is a checked type of input.

File: react/react1/week2/todo-list/src/TodoItem.jsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import React, { useState, useEffect } from "react";
+function TodoItem({ todo, onDelete }) {
+  const [isChecked, setIsChecked] = useState(false);
+
+  const changeOnChecked = (event) => {
+    setIsChecked(event.target.checked);
```
Comment (Divyajg) [inline react/react1/week2/todo-list/src/TodoItem.jsx:6]: `setIsChecked(!checked);`

**Review by Divyajg — APPROVED** [review]
Good job.
Try to use descriptive names like `todo `instead of `spanElement `.

File: react/react1/week2/todo-list/src/TodoItem.jsx (hunk 3 above)
Comment (agkabir) [reply react/react1/week2/todo-list/src/TodoItem.jsx:6]: yes. More smart.

File: react/react1/week2/todo-list/src/Counter.jsx (hunk 1 above)
Comment (agkabir) [reply react/react1/week2/todo-list/src/Counter.jsx:11]: thanks

**Comment by agkabir** [comment]
Thanks you for your nice comment.

**Pull request merged**